realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -49,23 +51,73 @@
  */
 public class RealmQuery<E extends RealmObject> {
 
-    private final Realm realm;
-    private final Table table;
-    private final LinkView view;
-    private final TableQuery query;
-    private final Map<String, Long> columns;
-    private final Class<E> clazz;
+    private BaseRealm realm;
+    private Class<E> clazz;
+    private String className;
+    private Map<String, Long> columns = new HashMap<String, Long>();
 
-    private static final String LINK_NOT_SUPPORTED_METHOD = "'%s' is not supported for link queries";
+    private Table table;
+    private LinkView view;
+    private TableQuery query;
 
     /**
-     * Creating a RealmQuery instance.
+     * Creates a query for objects of a given class from a {@link Realm}.
      *
      * @param realm  The realm to query within.
      * @param clazz  The class to query.
-     * @throws java.lang.RuntimeException Any other error.
+     * @return {@link RealmQuery} object. After building the query call one of the {@code find*} methods
+     * to run it.
+     */
+    public static <E extends RealmObject> RealmQuery<E> createQuery(Realm realm, Class<E> clazz) {
+        return new RealmQuery<E>(realm, clazz);
+    }
+
+    /**
+     * Creates a query for dynamic objects of a given type from a {@link DynamicRealm}.
+     *
+     * @param realm  The realm to query within.
+     * @param className  The type to query.
+     * @return {@link RealmQuery} object. After building the query call one of the {@code find*} methods
+     * to run it.
+     */
+    public static <E extends RealmObject> RealmQuery<E> createDynamicQuery(DynamicRealm realm, String className) {
+        return new RealmQuery<E>(realm, className);
+    }
+
+    /**
+     * Creates a sub-query from a existing {@link RealmResults}.
```

Comment:
I'm not sure like to use "sub-query" as a term. Currently, we don't have projections and cannot return values from aggregate functions. In SQL, subqueries have a well-defined meaning, and I am afraid that users will be confused.